compiler/test/dotty/tools/vulpix/ParallelTesting.scala

```diff
@@ -658,20 +658,48 @@ trait ParallelTesting extends RunnerOrchestration { self =>
           if (errors > 0)
             errorMap.put(s"${file.getPath}:${lineNbr}", errors)
 
+
           val noposErrors = line.sliding("// nopos-error".length).count(_.mkString == "// nopos-error")
           if (noposErrors > 0) {
             val nopos = errorMap.get("nopos")
             val existing: Integer = if (nopos eq null) 0 else nopos
             errorMap.put("nopos", noposErrors + existing)
           }
 
+          maybeWarning(line).foreach(warning => echo(s"Possible typo in file ${file.getCanonicalPath} at line number $lineNbr \n$warning"))
+
           expectedErrors += noposErrors + errors
         }
       }
 
       (errorMap, expectedErrors)
     }
 
+    /**
+     * We want to warn the developer about some of the possible typos when trying to point "error" and "nopos-error"
+     * For example - A warning could be emitted if the developer mistakenly types in "//error" instead of "// error"
+     * or "//nopos-error" instead of "// nopos-error"
+     *
+     * @param line the line which will be checked for possible typos
+     * @return Some(warningMessage) or None
+     */
+    def maybeWarning(line : String) : Option[String] = {
```

Comment:
It would be more efficient to inline these methods. There is no real need to box the message in an `Option`, we could directly echo the warning. The `line.contains` is executed more times than necessary, the `hasTypo` is an unnecessary check as `getWarning` also check those conditions.